crates/orchestrator/src/jobs/state_update_job/mod.rs

```diff
@@ -31,26 +44,105 @@ impl Job for StateUpdateJob {
         })
     }
 
-    async fn process_job(&self, _config: &Config, _job: &JobItem) -> Result<String> {
+    async fn process_job(&self, config: &Config, job: &JobItem) -> Result<String> {
         // Read the metadata to get the blocks for which state update will be performed.
-        // For each block, get the program output (from the PIE?) and the
-        todo!()
+        // We assume that blocks nbrs are formatted as follow: "2,3,4,5,6".
+        let blocks_to_settle = job.metadata.get(JOB_METADATA_STATE_UPDATE_BLOCKS_TO_SETTLE_KEY).ok_or_else(|| {
+            eyre!(eyre!("Blocks number to settle must be specified (prover job #{})", job.internal_id))
+        })?;
+        let block_numbers: Vec<String> = blocks_to_settle.split(',').map(String::from).collect();
+
+        // TODO: remove when SNOS is correctly stored in DB/S3
+        // Test metadata to fetch the snos output from the test folder
+        let fetch_from_tests = job.metadata.get(METADATA_FETCH_FROM_TESTS).map_or(false, |value| value == "TRUE");
+
```

Comment:
Added a validation function

Reply:
For me this would be the responsible of the worker to correct the input & send it corrected. What do you think? Happy to change if needed

Reply:
I think the worker can use `config.settlement_client.get_last_settled_block` to know which was the failing block & either create a new jobs for the remaining blocks / update the metadata

But you're right we can also create a new error type that carries the block number if an error occurs. Any preference?  I'd say using the settlement client is better as we're 100% sure about the information instead of trusting the failed job for the info

Reply:
Looks good, just a question
>We store this mapping inside of metadara

Currently metadata is a String -> String mapping - so that would not work right? @apoorvsadana

Reply:
Looks good, I'll implement this asap

Reply:
Mmmmh, one thing that's boring is that we only pass `job` as a reference not mutable. Meaning that we can't update it through the job. I can update to `job: &mut JobItem` but it also means that I'll have to update it everywhere. 

Any thoughts? @apoorvsadana @unstark 

I guess it could also be possible to set everything into a big string that has a certain format and return that as the external_id; so the caller that has the mutable job can update the metadata.

Reply:
I can think of two options

1. Pass `job: &mut JobItem` as you mentioned
2. Update the job inside `process_job` itself

I find (1) to be cleaner as it's only one DB call. Will it be a lot of changes? I think it should just changing the function args?

Reply:
I like (1), clean/easy and not much changes yes. Implemented it! Let me know if that's what you had in mind @apoorvsadana

Reply:
Looks good but can you also change the update logic here - https://github.com/karnotxyz/madara-orchestrator/blob/main/crates/orchestrator/src/jobs/mod.rs#L102-L102

Right now we only update external id and metadata. Let's change it so that in the general `process_job` function, after the job is processed we rewrite the Job to the DB because we don't know what changes would have been made to the mutable.

Reply:
Done!